src/main/javac/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaNodeToEntityConverter.java

```diff
@@ -0,0 +1,204 @@
+import org.apache.brooklyn.util.core.ResourceUtils;
+import org.apache.brooklyn.util.core.config.ConfigBag;
+import org.apache.brooklyn.util.core.flags.TypeCoercions;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.commons.lang3.StringUtils;
+import org.jclouds.compute.domain.OsFamily;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+import alien4cloud.model.components.AbstractPropertyValue;
+import alien4cloud.model.components.ImplementationArtifact;
+import alien4cloud.model.components.Interface;
+import alien4cloud.model.components.Operation;
+import alien4cloud.model.components.ScalarPropertyValue;
+import alien4cloud.model.topology.NodeTemplate;
+
+public class ToscaNodeToEntityConverter {
+
+    private static final Logger log = LoggerFactory.getLogger(ToscaNodeToEntityConverter.class);
+
+    private final ManagementContext mgnt;
+    private NodeTemplate nodeTemplate;
+    private String nodeId;
+
+    private ToscaNodeToEntityConverter(ManagementContext mgmt) {
+        this.mgnt = mgmt;
+    }
+
+    public static ToscaNodeToEntityConverter with(ManagementContext mgmt) {
+        return new ToscaNodeToEntityConverter(mgmt);
+    }
+
+    public ToscaNodeToEntityConverter setNodeTemplate(NodeTemplate nodeTemplate) {
+        this.nodeTemplate = nodeTemplate;
+        return this;
+    }
+
+    public ToscaNodeToEntityConverter setNodeId(String nodeId) {
+        this.nodeId = nodeId;
+        return this;
+    }
+
+    public EntitySpec<? extends Entity> createSpec() {
+        if (this.nodeTemplate == null) {
+            throw new IllegalStateException("TOSCA node template is missing. You must specify it by using the method #setNodeTemplate(NodeTemplate nodeTemplate)");
+        }
+        if (StringUtils.isEmpty(this.nodeId)) {
+            throw new IllegalStateException("TOSCA node ID is missing. You must specify it by using the method #setNodeId(String nodeId)");
+        }
+
+        EntitySpec<?> spec = null;
+
+        CatalogItem catalogItem = CatalogUtils.getCatalogItemOptionalVersion(this.mgnt, this.nodeTemplate.getType());
+        if (catalogItem != null) {
+            log.info("Found Brooklyn catalog item that match node type: " + this.nodeTemplate.getType());
+            spec = (EntitySpec<?>) this.mgnt.getCatalog().createSpec(catalogItem);
+        } else {
+            try {
+                log.info("Found Brooklyn entity that match node type: " + this.nodeTemplate.getType());
+                spec = EntitySpec.create((Class<? extends Entity>) Class.forName(this.nodeTemplate.getType()));
+            } catch (ClassNotFoundException e) {
+                log.info("Cannot find any Brooklyn catalog item nor Brooklyn entities that match node type: " +
+                        this.nodeTemplate.getType() + ". Defaulting to a VanillaSoftwareProcess");
+                spec = EntitySpec.create(VanillaSoftwareProcess.class);
```

Comment:
Can use `BasicEntity` for generic items. If it is supposed to have children then `BasicApplication` is a better fit because of the lifecycle mgmt. Both don't require a backing machine. If the goal is to spin up a machine then `MachineEntity` is the best fit.